scripts/generate-zod-schemas.js

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env node
+
+import { readFileSync, writeFileSync, mkdirSync, existsSync } from 'fs';
+import { join, dirname } from 'path';
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+const projectRoot = join(__dirname, '..');
+
+// Convert TypeScript type to Zod schema string
+function typeToZod(prop) {
+  const { type, enumValues, defaultValue } = prop;
+  let zodType = 'z.any()';
+
+  // Handle enum/union types first
+  if (enumValues && enumValues.length > 0) {
+    const enumString = enumValues.map(val => `'${val}'`).join(', ');
+    zodType = `z.enum([${enumString}])`;
+  }
+  // Handle union types with numbers (like 0 | 1 | 2 | 3)
+  else if (type.includes('|') && /\d/.test(type)) {
+    const values = type.split('|').map(v => v.trim());
+    const allNumbers = values.every(v => !isNaN(Number(v)));
+    if (allNumbers) {
+      zodType = `z.union([${values.map(v => `z.literal(${v})`).join(', ')}])`;
+    }
+  }
+  // Handle basic types
+  else if (type === 'boolean') {
+    zodType = 'z.boolean()';
+  }
+  else if (type === 'string') {
+    zodType = 'z.string()';
+  }
+  else if (type === 'number') {
+    zodType = 'z.number()';
+  }
+  else if (type === 'Date') {
+    zodType = 'z.date()';
+  }
+  // Handle array types
+  else if (type.includes('[]') || type.includes('Array<')) {
+    zodType = 'z.array(z.any())';
+  }
+  // Handle React types
+  else if (type.includes('ReactNode') || type.includes('React.ReactNode')) {
+    zodType = 'z.any()';
+  }
+  else if (type.includes('ReactElement') || type.includes('React.ReactElement')) {
+    zodType = 'z.any()';
+  }
+  else if (type.includes('MouseEvent') || type.includes('KeyboardEvent') || type.includes('Event')) {
+    zodType = 'z.any()';
+  }
```

Comment:
Returning anys for these also seems suboptimal, are there not better zod types we can give it?

If not I would probably request refactoring this to do a .some or something instead of chaining else/ifs here too.